### PR TITLE
Updated PHPUnit to version 7 + gh-24 + gh-42 + gh-85 + gh-83 + gh-71

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: src/tarantool_schema.c [hunk 1]
```diff
@@ -14,27 +14,41 @@
...
-		const struct schema_index_value **lval,
-		const struct schema_index_value **rval,
-		void *arg
-) {
-	(void )arg;
-	if ((*lval)->key.id_len != (*rval)->key.id_len)
-		return 0;
-	return !memcmp((*lval)->key.id, (*rval)->key.id, (*rval)->key.id_len);
-}
+#define MH_DEFINE_CMPFUNC(NAME, TYPE)								\
```
Comment (Totktonada) [inline src/tarantool_schema.c]: I trying to retain myself from formatting nitpicking, but here the entire block formatted with exceeding 80 symbols limit.

File: src/tarantool.c [hunk 2]
```diff
@@ -695,6 +744,11 @@ int get_indexno_by_name(tarantool_connection *obj, int space_no,
 		THROW_EXC("Index ID must be String or Long");
 		return FAILURE;
 	}
+
+	if (obtain_space_by_spaceno(obj, space_no) == FAILURE) {
```
Comment (Totktonada) [inline src/tarantool.c]: I think this call should be a part of get_spaceno_by_name.

File: README.md [hunk 3]
```diff
@@ -155,8 +154,10 @@ Tarantool class instance
 
 ``` php
 $tnt = new Tarantool(); // -> new Tarantool('localhost', 3301);
-$tnt = new Tarantool('tarantool.org'); // -> new Tarantool('tarantool.org', 3301);
-$tnt = new Tarantool('localhost', 16847);
+$tnt = new Tarantool('tcp://test:test@localhost');
+$tnt = new Tarantool('tcp://test:test@localhost:3301');
+$tnt = new Tarantool('test:test@unix/:/var/tmp/tarantool.sock);
```
Comment (Totktonada) [inline README.md]: Typo: unmatched quote.

File: README.md [hunk 4]
```diff
@@ -155,8 +154,10 @@ Tarantool class instance
...
+$tnt = new Tarantool('unix:///var/tmp/tarantool.sock); /* if no login is needed */
```
Comment (Totktonada) [inline README.md]: Typo: unmatched quote.

File: src/tarantool_url.c [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+		return NULL;
+	char *rv = pecalloc(sizeof(char), part_len + 1, persistent);
+	rv[part_len] = '\0';
+	memcpy(rv, part, part_len);
+	return (const char *)rv;
+}
+
+struct tarantool_url *
+tarantool_url_parse(const char *url, int persistent) {
+	struct uri parsed; memset(&parsed, 0, sizeof(struct uri));
```
Comment (Totktonada) [inline src/tarantool_url.c]: Nit: I would avoid two operators on the same line.

File: src/tarantool_url.c [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+		if (parsed.service_len == 0) parsed.service = NULL;
+		if (parsed.path_len == 0)    parsed.path    = NULL;
+
+		/* If both are set or not then throw error */
+		if ((parsed.service == NULL) == (parsed.path == NULL))
+			return NULL;
+
+		path     = parsed.service ? parsed.service     : parsed.path;
+		path_len = parsed.service ? parsed.service_len : parsed.path_len;
```
Comment (Totktonada) [inline src/tarantool_url.c]: It seems that `service` is not the same as `path`. Consider test/unit/uri.c in tarantool repository:

```
  6 test(const char *s, const char *scheme, const char *login, const char *password,           
  7      const char *host, const char *service, const char *path,
  8      const char *query, const char *fragment, int host_hint)
  9 {  

 82         test("host:service/path1/path2/path3", NULL, NULL, NULL, "host",                   
 83              "service", "/path1/path2/path3", NULL, NULL, 0);
```

File: README.md [hunk 7]
```diff
@@ -114,7 +114,7 @@ _**Description**_: Available Tarantool Constants
 
 ``` php
 Tarantool {
-    public       Tarantool::__construct ( [ string $host = 'localhost' [, int $port = 3301 [, string $user = "guest" [, string $password = NULL [, string $persistent_id = NULL ] ] ] ] ] )
+    public       Tarantool::__construct ( [ string $uri = 'tcp://guest@localhost:3301' [, string $persistent_id = NULL ] ] )
```
Comment (Totktonada) [inline README.md]: Default port is 3303 as I see in src/tarantool_url.c.

File: src/tarantool_url.c (hunk 6 above)
Comment (Totktonada) [reply src/tarantool_url.c]: I understand it with the following test case from the same file:

```
202         test("unix/:/path1/path2/path3:/p1/p2?q1=v1&q2=v2#fragment",                       
203              NULL, NULL, NULL, "unix/", "/path1/path2/path3", "/p1/p2",                    
204              "q1=v1&q2=v2", "fragment", 3);  
```

`service` is path to a unix socket file and `path` is some logical (say, http) path.

But this is only for `unix/:/path` URIs. When we use `unix:///path` the path is placed in `path`.

So the implementation is correct.

File: src/tarantool.c [hunk 8]
```diff
@@ -225,7 +225,7 @@ static int __tarantool_connect(tarantool_object *t_obj) {
 	TSRMLS_FETCH();
 	tarantool_connection *obj = t_obj->obj;
 	int status = SUCCESS;
-	long count = TARANTOOL_G(retry_count);
+	long count = TARANTOOL_G(retry_count) + 1;
```
Comment (Totktonada) [inline src/tarantool.c]: So in case of `retry_count` > 0 the real retry count will be `retry_count` + 1? I think it should be handled in some other way.

**Comment by Totktonada** [comment]
The first review is done. @bigbes, the ball is on your court.

**Comment by bigbes** [comment]
Superseded by #140 #141 #142 #143

**Pull request closed without merging**